android/src/main/java/com/reactlibrary/QonversionModule.java

```diff
@@ -65,16 +67,31 @@ public String getName() {
     }
 
     @ReactMethod
-    public void storeSDKInfo(String sourceKey, String source, String sdkVersionKey, String sdkVersion) {
-        SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(getCurrentActivity().getApplication()).edit();
+    public void storeSDKInfo(String sourceKey, String source, String sdkVersionKey, String sdkVersion,final Promise promise) {
+        Activity currentActivity = getCurrentActivity();
+        if(currentActivity == null){
+            QonversionError qonversionError = new QonversionError(QonversionErrorCode.UnknownError, "Android current activity is null, cannot perform the process.");
+            promise.reject(qonversionError.getCode().toString(),qonversionError.getDescription());
+            return;
+        }
+
+        SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(currentActivity.getApplication()).edit();
         editor.putString(sdkVersionKey, sdkVersion);
         editor.putString(sourceKey, source);
         editor.apply();
+        promise.resolve(null);
     }
 
     @ReactMethod
     public void launchWithKey(String key, Boolean observeMode, final Promise promise) {
-        Qonversion.launch(getCurrentActivity().getApplication(), key, observeMode, new QonversionLaunchCallback()
+        Activity currentActivity = getCurrentActivity();
+        if(currentActivity == null){
+            QonversionError qonversionError = new QonversionError(QonversionErrorCode.UnknownError, "Android current activity is null, cannot perform the process.");
+            promise.reject(qonversionError.getCode().toString(),qonversionError.getDescription());
+            return;
+        }
+
+        Qonversion.launch(currentActivity.getApplication(), key, observeMode, new QonversionLaunchCallback()
```

Comment:
As I said, will be good to add a check of locally stored SDK info here and store it to `SharedPreferences` if `currentActivity` is not null. Also will be good to add a separate function that will save SDK info to preferences. You can use it here and inside the `storeSDKInfo` function.

Reply:
Done!